### PR TITLE
feat: Add 5.2 bug fixes and minor changes

### PR DESCRIPTION
**Pull request by tylerkelly13**

### CODE REVIEW

File: release-notes/release-notes52.md [hunk 1]
```diff
@@ -83,6 +84,26 @@ A new `link_default_protocol` option has been added to the Autolink and Link plu
...
 
+## Minor changes for TinyMCE 5.2
+
+{{site.productname}} 5.2 introduces the following minor changes:
+
+- Adds the ability to apply formats to spaces.
+- Adds a drop shadow below the toolbar while in sticky mode and introduced Oxide variables to customize it when creating a custom skin.
+- Adds a `quickbars_image_toolbar` setting to allow for the image quickbar to be turned off.
+- Adds the `loading` attribute to the default schema for `iframe` and `img` . Patch contributed by [ataylor32](https://github.com/ataylor32).
+- Adds new `getNodeFilters` and `getAttributeFilters` functions to the `editor.serializer` instance.
```
Comment (metricjs) [inline release-notes/release-notes52.md]: Not sure of the answer but: instance or API?

File: release-notes/release-notes52.md [hunk 2]
```diff
@@ -83,6 +84,26 @@ A new `link_default_protocol` option has been added to the Autolink and Link plu
...
+- Adds error message events that fire when a resource loading error occurs.
```
Comment (metricjs) [inline release-notes/release-notes52.md]: Might wanna specify if error banner or console error

File: release-notes/release-notes52.md [hunk 3]
```diff
@@ -83,6 +84,26 @@ A new `link_default_protocol` option has been added to the Autolink and Link plu
...
+- Changes the default schema to disallow `onchange` for select elements.
```
Comment (metricjs) [inline release-notes/release-notes52.md]: This was for security reasons iirc? Would we wanna say that?

File: release-notes/release-notes52.md [hunk 4]
```diff
@@ -83,6 +84,26 @@ A new `link_default_protocol` option has been added to the Autolink and Link plu
...
+- Changes iframe mode to set selection on content init if selection doesn't exist.
+- Changes table related icons to align them with the visual style of the other icons.
+- Changes and improved the visual appearance of the color input field.
```
Comment (metricjs) [inline release-notes/release-notes52.md]: "Changes and improved"?

File: release-notes/release-notes52.md [hunk 5]
```diff
@@ -125,10 +146,37 @@ The {{site.productname}} 5.2 release includes **Spellchecker Pro** 2.0.
...
+- Fixed the `file_picker_callback` functionality for the Image, Link and Media plugins.
+- Fixed an issue where the floating toolbar drawer would sometimes break if the editor is resized while the drawer is open.
+- Fixed an incorrect `external_plugins` loading error message.
+- Fixed the resize handler not being hidden for ARIA purposes. Patch contributed by [Parent5446](https://github.com/Parent5446).
+- Fixed an issue where content could be lost if a misspelled word was selected and spellchecking was disabled.
+- Fixed validation errors in the CSS where certain properties had the wrong default value.
+- Fixed an issue where forced root block attributes were not applied when removing a list.
+- Fixed an issue where the element path was not cleared when there are no parents.
+- Fixed an issue where width and height in svg icons containing `rect` elements were overridden by the CSS reset.
+- Fixed an issue where uploading images with `images_reuse_filename` enabled and that included a query parameter would generate an invalid URL.
```
Comment (metricjs) [inline release-notes/release-notes52.md]: "enabled and that" <- something feels odd

File: release-notes/release-notes52.md [hunk 6]
```diff
@@ -83,6 +84,26 @@ A new `link_default_protocol` option has been added to the Autolink and Link plu
 - For information on using this setting with the Autolink plugin, see: [Autolink plugin - `link_default_protocol`]({{site.baseurl}}/plugins/autolink/#link_default_protocol).
 - For information on using this setting with the Link plugin, see: [Link plugin - `link_default_protocol`]({{site.baseurl}}/plugins/link/#link_default_protocol).
 
+## Minor changes for TinyMCE 5.2
```
Comment (lnewson) [inline release-notes/release-notes52.md]: The following are really security fixes, so not sure if you want to remove them and just put something more generic?
```
- Changes the default schema to disallow `onchange` for select elements.
- Changes the editor parser to treat CDATA nodes as bogus HTML comments to match the HTML parsing specification. A new `preserve_cdata` setting has been added to preserve CDATA nodes if required
```
Alternatively, maybe we could just add another section saying some security issues were fixed?

File: release-notes/release-notes52.md [hunk 7]
```diff
@@ -125,10 +146,37 @@ The {{site.productname}} 5.2 release includes **Spellchecker Pro** 2.0.
...
+- Fixed an issue where the element path was not cleared when there are no parents.
+- Fixed an issue where width and height in svg icons containing `rect` elements were overridden by the CSS reset.
+- Fixed an issue where uploading images with `images_reuse_filename` enabled and that included a query parameter would generate an invalid URL.
+- Fixed the `closeButton` property not working when opening notifications.
+- Fixed keyboard flicker when opening a context menu on mobile.
+- Fixed issue where plus icon svg contained strokes.
 
 ## Deprecated features
 
+The `toolbar_drawer` option has been deprecated with the release of {{site.productname}} 5.2. The `toolbar_drawer` option was renamed to `toolbar_mode`. This change was made to reflect the range of settings available for this option.
```
Comment (metricjs) [inline release-notes/release-notes52.md]: "range of settings" -> should we imply that we added new options there, to better justify it?

File: release-notes/release-notes52.md [hunk 8]
```diff
@@ -125,10 +146,37 @@ The {{site.productname}} 5.2 release includes **Spellchecker Pro** 2.0.
 
 {{site.productname}} 5.2 provides fixes for the following bugs:
 
--
+- Fixed the incorrect parsing of malformed/bogus HTML comments.
+- Fixed the `quickbars` selection toolbar appearing on non-editable elements.
+- Fixed a bug with alignment toolbar buttons sometimes not changing state correctly.
+- Fixed the `codesample` toolbar button not toggling when selecting code samples other than HTML.
+- Fixed content incorrectly scrolling to the top or bottom when pressing enter when the content was already in view.
```
Comment (lnewson) [inline release-notes/release-notes52.md]: `if the content was already in view`?

File: release-notes/release-notes52.md [hunk 9]
```diff
@@ -125,10 +146,37 @@ The {{site.productname}} 5.2 release includes **Spellchecker Pro** 2.0.
...
+- Fixed the `scrollIntoView` event potentially hiding elements behind the toolbar.
```
Comment (lnewson) [inline release-notes/release-notes52.md]: `scrollIntoView` isn't an event, it's an editor API.

File: release-notes/release-notes52.md [hunk 10]
```diff
@@ -125,10 +146,37 @@ The {{site.productname}} 5.2 release includes **Spellchecker Pro** 2.0.
...
+- Fixed the editor not respecting the `resize_img_proportional` setting due to legacy code.
+- Fixed the floating toolbar drawer flickering in inline mode.
+- Fixed an issue where the `template` plugin dialog would be indefinitely blocked on a failed template load.
+- Fixed the `mscontrolselect` event not being unbound on the Microsoft Internet Explorer and Microsoft Edge browsers.
+- Fixed the "Confirm" dialog footer buttons to only highlight the "Yes" button.
```
Comment (lnewson) [inline release-notes/release-notes52.md]: confirm is a type of dialog, so should it be in quotes?

File: release-notes/release-notes52.md [hunk 11]
```diff
@@ -83,6 +84,26 @@ A new `link_default_protocol` option has been added to the Autolink and Link plu
 - For information on using this setting with the Autolink plugin, see: [Autolink plugin - `link_default_protocol`]({{site.baseurl}}/plugins/autolink/#link_default_protocol).
 - For information on using this setting with the Link plugin, see: [Link plugin - `link_default_protocol`]({{site.baseurl}}/plugins/link/#link_default_protocol).
 
+## Minor changes for TinyMCE 5.2
+
+{{site.productname}} 5.2 introduces the following minor changes:
+
+- Adds the ability to apply formats to spaces.
+- Adds a drop shadow below the toolbar while in sticky mode and introduced Oxide variables to customize it when creating a custom skin.
```
[the remark this answers is not in the document]
Comment (lnewson) [reply release-notes/release-notes52.md]: I'd be concerned customers would interpret that as TinyMCE settings, so I think Oxide is required here for clarity.

File: release-notes/release-notes52.md [hunk 12]
```diff
@@ -83,6 +84,26 @@ A new `link_default_protocol` option has been added to the Autolink and Link plu
...
+- Adds a `quickbars_image_toolbar` setting to allow for the image quickbar to be turned off.
```
[the remark this answers is not in the document]
Comment (lnewson) [reply release-notes/release-notes52.md]: This is actually what was included in the changelog, so we probably should have fixed this before 5.2.0 was released :(

File: release-notes/release-notes52.md (hunk 1 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: API?

File: release-notes/release-notes52.md (hunk 2 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: that's correct. they are events that return a message.

File: release-notes/release-notes52.md (hunk 3 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: - Improved security by changing the default schema to disallow `onchange` for select elements.

File: release-notes/release-notes52.md (hunk 5 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: - Fixed an issue where uploading images with a query parameter and the `images_reuse_filename` setting enabled would generate an invalid URL.

File: release-notes/release-notes52.md (hunk 1 above)
Comment (lnewson) [reply release-notes/release-notes52.md]: It's technically both :man_shrugging: though I'd agree API would be better.

File: release-notes/release-notes52.md (hunk 7 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: updated

File: release-notes/release-notes52.md (hunk 6 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: changed the description a little to highlight that they are security related. Unwilling to pull them out into their own section at this time.

File: release-notes/release-notes52.md (hunk 8 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: fixed

File: release-notes/release-notes52.md (hunk 9 above)
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: fixed

File: release-notes/release-notes52.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (tylerkelly13) [reply release-notes/release-notes52.md]: retaining? 'Oxide'

**Pull request merged**